src/server/import-controller.js

```diff
@@ -0,0 +1,41 @@
+import PageRenderer from './page-renderer.js';
+import permissions from '../domain/permissions.js';
+import ServerConfig from '../bootstrap/server-config.js';
+import ImportService from '../services/import-service.js';
+import ExportApiClient from '../services/export-api-client.js';
+import { validateQuery } from '../domain/validation-middleware.js';
+import needsPermission from '../domain/needs-permission-middleware.js';
+import { getImportsQuerySchema } from '../domain/schemas/import-schemas.js';
+
+class ImportController {
+  static get inject() { return [ImportService, ExportApiClient, ServerConfig, PageRenderer]; }
+
+  constructor(importService, exportApiClient, serverConfig, pageRenderer) {
+    this.exportApiClient = exportApiClient;
+    this.importService = importService;
+    this.serverConfig = serverConfig;
+    this.pageRenderer = pageRenderer;
+  }
+
+  registerPages(app) {
+    app.get('/import', needsPermission(permissions.MANAGE_IMPORT), (req, res) => {
+      const initialState = { importSources: this.serverConfig.importSources };
+      return this.pageRenderer.sendPage(req, res, 'edit-bundle', 'import', initialState);
+    });
+  }
+
+  registerApi(router) {
+    router.get('/api/v1/imports', [needsPermission(permissions.MANAGE_IMPORT), validateQuery(getImportsQuerySchema)], async (req, res) => {
+      const importSource = {
+        name: req.query.importSourceName,
+        baseUrl: req.query.importSourceBaseUrl,
+        apiKey: req.query.importSourceApiKey
```

Comment:
We should not pass the API Key through the frontend. I would suggest that the frontend and controller layers only know the import sources by name. The API key and base URL can be looked up by the service (or even the client itself).

Reply:
Good point, I removed the API key from the  `GET /import` and removed both API key and base URL from `GET /api/v1/imports`.
Since the `import-controller` needs to use `serverConfig.importSources` to get the sources names anyway, I though it simpler to also then pass the base URL to the page.
And since the `import-controller needs to use `serverConfig.importSources` for the above case anyway, I thought it easier to also get the import source by name on this level and pass it to the `importService`

Reply:
Cool!